### PR TITLE
Added model function for ooo requests

### PR DESCRIPTION
**Pull request by sahsisunny**
**Description:** 
In this pull request, a new model function has been added to the OOO Request API. 

**Test Coverage:**
| File                 | Coverage | Statements | Branches | Functions | Lines   | Details  |
|----------------------|----------|------------|----------|-----------|---------|----------|
| oooRequests.ts       | 88.23    | 100        | 100      | 86.66     | 38-39   |

### CODE REVIEW

File: constants/oooRequest.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const OOO_REQUEST_ALREADY_PENDING =
+    "You already have a pending request. Please wait for it to be approved or rejected.";
+export const ERROR_WHILE_CREATING_OOO_REQUEST = "Error while creating ooo request";
```
Comment (iamitprakash) [inline constants/oooRequest.ts:3]: OOO

File: models/oooRequests.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      userId,
+      type,
+      from,
+      until,
+      message,
+      state: REQUEST_STATE.PENDING,
+      createdAt: Date.now(),
+      updatedAt: Date.now(),
+    };
+    const doc = await oooRestModel.add(request);
```
Comment (iamitprakash) [inline models/oooRequests.ts:31]: what is this...?

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: constants/oooRequest.ts (hunk 1 above)
Comment (sahsisunny) [reply constants/oooRequest.ts]: Fixed

File: models/oooRequests.ts (hunk 2 above)
Comment (sahsisunny) [reply models/oooRequests.ts]: Response doc object of the model

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**